### DISCUSSION TITLE
Zephyr API ( flash_stm32_check_status ) not giving correct values for failure

### DISCUSSION THREAD

**Discussion by bhardadinesh-eaton** [zephyrproject-rtos/zephyr#58881] [Q&A]
I am trying to use Zephyr API ( flash_stm32_check_status ( flash_stm32.c/.h ) for my development. I have used different Zephyr APIs for Read/Write/Erase, All working fine. Based on that, I want to get the status of any errors for Write Failure, Read Failure, Program Sequence Errors using flash_stm32_check_status but i am always getting 0 values for H743 controller. Any suggestion for this API is most welcome.

**Comment by erwango** [comment]
Note that `flash_stm32_check_status` is not an API but an internal driver function, called in Read/Write/Erase operations.
Then, it seems that `flash_stm32_check_status` clears errors when run, so calling it from the application after it is called from Read/Write/Erase function will indeed mostly always return 0 (unless for some reason the error flag can't be cleared).

**Reply by bhardadinesh-eaton** [reply]
@erwango  Thank you! Is there any API or function available to get the status of any failure?

**Reply by erwango** [reply]
Not that I'm aware of.

**Reply by bhardadinesh-eaton** [reply]
ok. I am looking for below errors for STM32 Controller,
![image](https://github.com/zephyrproject-rtos/zephyr/assets/95752525/edfebcb5-87db-4f19-811a-891f6bfc4ed5)

Definitely, I can write code to get the errors but, I have used most of Zephyr APIs and i am expecting if any zephyr API or function is available. We can used the same.

**Reply by erwango** [reply]
This would require an update of Zephyr FLASH API.
For now flash errors are cleared as part of existing flash API calls (write/read/erase).
A new get_error API can be added, but it requires to remove error clearing from existing driver implementation and get existing APIs users to update their application to systematically clear errors after API calls (which would also be a new API call to add).
You can have a try to propose this API behavior change but this is a significant impact for out of tree users.

**Reply by bhardadinesh-eaton** [reply]
@erwango  ok

cc : @SadanandPenore-eaton
